src/drive/ducks/files/FileOpener.js

```diff
@@ -18,55 +16,34 @@ class FileOpener extends Component {
   state = { url: null, loading: false, closing: false }
 
   componentWillMount() {
-    this.openFile()
+    this.redirectToFileViewer()
   }
 
-  async openFile() {
+  async redirectToFileViewer() {
     const { router, params: { fileId }, alert } = this.props
+
     try {
-      const url = await getFileDownloadUrl(fileId)
-      this.setState({ url })
+      const fileInfo = await cozy.client.files.statById(fileId)
+
+      const { attributes: { dir_id } } = fileInfo
+      // Go to the parent folder
+      router.push(`/folder/${dir_id}/file/${fileId}`)
     } catch (e) {
+      console.warn(e)
       router.push('/')
       alert({
         message: 'alert.could_not_open_file'
       })
-    } finally {
-      this.setState({ loading: false })
     }
   }
 
-  handleClose = async () => {
-    this.setState({ closing: true })
-    const { router, params: { fileId } } = this.props
-    const { relationships: { parent } } = await cozy.client.files.statById(
-      fileId
-    )
```

Comment:
I stumbled upon a case where `relationships.parent` was `undefined` whereas dir_id was defined. This is why I use `dir_id` [above](https://github.com/cozy/cozy-drive/pull/691/files#diff-cec592dc3fb6c108dee8815cc445e469R28). Any idea why ?

Reply:
Not really. @jinroh can you think of a case where that would happen?